instrumentation/log4j/log4j-appender-1.2/javaagent/src/test/groovy/Log4j1Test.groovy

```diff
@@ -93,6 +94,8 @@ class Log4j1Test extends AgentInstrumentationSpecification {
     severityText = args[2]
   }
 
+  // this used to be a problem in early Java 17 and Java 18 versions too
+  @IgnoreIf({ System.getProperty("java.version").startsWith("19") })
```

Comment:
As this is a reoccurring problem with each new java version WDYT about working around it with something like 
```
  static {
    org.apache.log4j.helpers.Loader.java1 = false
  }
```
Basically the issue is that log4j tries to figure out whether it is running on java1, presumably because `ThreadLocal` was added in java2, by parsing `java.version`. `19` gets flagged as java1 so mdc will be unavailable. Next version which would be something like `19.0.1` would work again.

Reply:
😱 thank you!! I have been wondering the cause of this, but never was able to track it down 💯